Advanced-Recruit-Training/src/BatteryStateOfChargeService.h

```diff
@@ -23,6 +23,15 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
 private:
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
+    double initialAhUsed_;
+    double chargeAh_;
+    double averageCurrent_;
```

Comment:
Consider the lifetime of this variable: Its value isn't used outside of `addData` and the value doesn't carry over into other calls of `addData`. With that in mind, it's probably best for it to be a local variable.